.github/workflows/main.yml

```diff
@@ -0,0 +1,58 @@
+name: Main workflow
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          #- macos-latest
+          #- windows-latest
+        ocaml-version:
+          - 4.09.1
+          - 4.08.1
+          - 4.07.1
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Use OCaml ${{ matrix.ocaml-version }}
+        uses: avsm/setup-ocaml@v1
+        with:
+          ocaml-version: ${{ matrix.ocaml-version }}
+
+      - name: Pinning BAP
+        run: opam pin add bap . --no-action
+
+      - name: Configuring Homebrew LLVM
+        if: matrix.os == 'macos-latest'
+        run: echo '::set-env name=LLVM_CONFIG::/usr/local/opt/llvm/bin/llvm-config'
+
+      - name: Installing system dependencies
+        run: opam depext bap --yes --with-doc --with-test
+
+      - name: Installing opam dependencies
+        run: opam install . --deps-only --with-doc --with-test
```

Comment:
in order to run tests during the installation, keep `--with-tests`

and add to opam/opam's install field the following
```
[make "test"] {with-test}
```